napari/_qt/layer_controls/qt_tracks_controls.py

```diff
@@ -124,10 +124,20 @@ def _on_head_length_change(self):
     def _on_properties_change(self):
         """Change the properties that can be used to color the tracks."""
         with self.layer.events.properties.blocker():
+            pre_store = self.layer.properties_to_color_by
 
             with qt_signals_blocked(self.color_by_combobox):
                 self.color_by_combobox.clear()
-            self.color_by_combobox.addItems(self.layer.properties_to_color_by)
+
+            if pre_store in self.layer.properties_to_color_by:
+                self.layers.color_by = pre_store
+                self.color_by_combobox.addItems(
+                    self.layer.properties_to_color_by
+                )
+            else:
+                self.color_by_combobox.addItems(
+                    self.layer.properties_to_color_by
+                )
```

Comment:
This looks equivalent and shorter/simpler to read.

```suggestion
            if pre_store in self.layer.properties_to_color_by:
                self.layers.color_by = pre_store
            self.color_by_combobox.addItems(self.layer.properties_to_color_by)
```

Looking at it now, I think the issue here is likely being cause by `addItems`, which I think will change the current item of qcombobox to be the first one added here. If that's true then we probably want to block that by using the `qt_signal_blocker` above and then ensuring that `self.layer.color_by` is the current item.

Reply:
Another good resource to look at are the Qt docs on `QComboBox`: https://doc.qt.io/qt-5/qcombobox.html#details

Note that the following line in our `qt_tracks_controls.py` means that we react to the `currentTextChanged` signal:

```python
self.color_by_combobox.currentTextChanged.connect(self.change_color_by)
```

And I'm pretty sure that calling `QComboBox.clear` followed by `QComboBox.addItems` (as we do in the code here) means that signal will be emitted (likely making the first item the current one).